src/Spice86.Core/Emulator/InterruptHandlers/Dos/DosInt21Handler.cs

```diff
@@ -376,44 +380,126 @@ public void CloseFile(bool calledFromVm) {
     /// </returns>
     /// <param name="calledFromVm">Whether this was called by the emulator.</param>
     public void CreateFileUsingHandle(bool calledFromVm) {
-        string fileName = GetZeroTerminatedStringAtDsDx();
+        string fileName = _dosStringDecoder.GetZeroTerminatedStringAtDsDx();
         ushort fileAttribute = State.CX;
         if (LoggerService.IsEnabled(LogEventLevel.Verbose)) {
-            LoggerService.Verbose("CREATE FILE USING HANDLE: {FileName} with attribute {FileAttribute}", fileName, fileAttribute);
+            LoggerService.Verbose("CREATE FILE USING HANDLE: {FileName} with attribute {FileAttribute}",
+                fileName, fileAttribute);
         }
         DosFileOperationResult dosFileOperationResult = _dosFileManager.CreateFileUsingHandle(fileName, fileAttribute);
         SetStateFromDosFileOperationResult(calledFromVm, dosFileOperationResult);
     }
 
+    /// <summary>
+    /// Implements the INT21H function 0x3D, which reads standard input DOS Device and outputs it to the standard output DOS Device. <br/>
+    /// </summary>
+    /// <remarks>
+    /// TODO: Add check for Ctrl-C and Ctrl-Break in STDIN, and call INT23H if it happens.
+    /// </remarks>
+    public void BufferedInput() {
+        uint address = MemoryUtils.ToPhysicalAddress(State.DS, State.DX);
+        DosInputBuffer inputBufferHolder = new DosInputBuffer(Memory, address);
+        int readCount = 0;
+        if (!_dosFileManager.TryGetStandardInput(out CharacterDevice? standardInput) ||
```

Comment:
Not sure to understand well, those are stdin and stdout (or rather the device mapped there right?)